controller/lib/hal/hal.h

```diff
@@ -54,11 +54,27 @@ limitations under the License.
 
 #else // !TEST_MODE
 
-#if !defined(ARDUINO_AVR_UNO) && !defined(ARDUINO_NUCLEO_L452RE)
+#if !defined(ARDUINO_AVR_UNO) && !defined(ARDUINO_NUCLEO_L452RE) and !defined(BARE_STM32)
 #error                                                                         \
     "When running without TEST_MODE, expecting ARDUINO_AVR_UNO or ARDUINO_NUCLEO_L452RE to be defined"
 #endif
+
+#if defined(BARE_STM32)
+#include <stdint.h>
+
+// Some constants that would normally be in the Arduino header
```

Comment:
It's some uglyness in the hal.h file that I didn't want to mess with yet.  Once we remove the Arduino builds it can be fixed in another PR.
Basically, the problem is the way the PinMode enum is defined.  It uses these constants which it expects to be defined in the arduino headers.  There's no good reason for that and it should really be cleaned up.
For now I just add these few constants so the code will compile.